src/StaticWebAssetsSdk/Tasks/RewriteCss.cs

```diff
@@ -2,60 +2,481 @@
+            {
+                if (item.Children.Count >= 2
+                    && item.Children[0] is TokenItem firstChild
+                    && firstChild.TokenType == CssTokenType.At
+                    && item.Children[1] is TokenItem secondChild
+                    && string.Equals(secondChild.Text, "import", StringComparison.OrdinalIgnoreCase))
+                {
+                    var location = _sourceFile.GetLocation(item.Start);
+                    _diagnostics.Add(new(ImportNotAllowedErrorMessage, _filePath, location.Line, location.Character));
+                }
+
+                base.VisitAtDirective(item);
+            }
+        }
+
+        private class Visitor
+        {
+            private readonly ComplexItem _root;
+
+            public Visitor(ComplexItem root)
+            {
+                _root = root ?? throw new ArgumentNullException(nameof(root));
+            }
+
+            public void Visit()
+            {
+                VisitDefault(_root);
+            }
+
+            protected virtual void VisitSelector(Selector item)
+            {
+                VisitDefault(item);
+            }
+
+            protected virtual void VisitAtDirective(AtDirective item)
+            {
+                VisitDefault(item);
+            }
+
+            protected virtual void VisitDeclaration(Declaration item)
+            {
+                VisitDefault(item);
+            }
+
+            protected virtual void VisitDefault(ParseItem item)
+            {
+                if (item is ComplexItem complexItem)
+                {
+                    VisitDescendants(complexItem);
+                }
+            }
+
+            private void VisitDescendants(ComplexItem container)
+            {
+                foreach (var child in container.Children)
+                {
+                    switch (child)
+                    {
+                        case Selector selector:
+                            VisitSelector(selector);
+                            break;
+                        case AtDirective atDirective:
+                            VisitAtDirective(atDirective);
+                            break;
+                        case Declaration declaration:
+                            VisitDeclaration(declaration);
+                            break;
+                        default:
+                            VisitDefault(child);
+                            break;
+                    }
+                }
+            }
+        }
+
+        private abstract class CssEdit
+        {
+            public int Position { get; set; }
+        }
+
+        private class InsertSelectorScopeEdit : CssEdit
+        {
+        }
+
+        private class InsertKeyframesNameScopeEdit : CssEdit
+        {
+        }
+
+        private class DeleteContentEdit : CssEdit
+        {
+            public int DeleteLength { get; set; }
+        }
+
+        private class SourceFile
```

Comment:
I added this class to provide a helper for extracting line/column numbers from `@import` uses. We could definitely remove this if we're okay with not having line numbers in error messages. If we do that, we'll probably want to deduplicate error messages that relate to the same file.